csi/controller.go

```diff
@@ -887,6 +887,34 @@ func (s *OsdCsiServer) CreateSnapshot(
 		return nil, status.Error(codes.InvalidArgument, "Name must be provided")
 	}
 
+	// Get secret if any was passed
+	ctx = s.setupContext(ctx, req.GetSecrets())
+	ctx, cancel := grpcutil.WithDefaultTimeout(ctx)
+	defer cancel()
+
+	// Get any labels passed in by the CO
+	_, locator, _, err := s.specHandler.SpecFromOpts(req.GetParameters())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Unable to get parameters: %v", err)
+	}
+	fmt.Printf("locator.VolumeLabels: %s", locator.VolumeLabels)
+	// Check ID is valid with the specified volume capabilities
+	snapshotType, ok := locator.VolumeLabels["snapshotType"]
```

Comment:
Can you externalize this to be a constant?

Also, let's name this with our standard convention: `csi.openstoarge.org/snapshot-type`.

See this area for an example of the prefix:
https://github.com/libopenstorage/openstorage/blob/master/csi/controller.go#L42-L46